csm/core/services/firmware_update.py

```diff
@@ -30,7 +30,7 @@ class FirmwareUpdateService(UpdateService):
     def __init__(self, provisioner, storage_path, update_repo):
         super().__init__(provisioner, update_repo)
         self._fw_storage_path = storage_path
-        os.makedirs(storage_path, exist_ok=True)
+        os.makedirs(self._fw_storage_path, exist_ok=True)
```

Comment:
@udayan-y21 Is it required to create the folder during initialization? Expectation is to create the folder during upload if not already created.
@kupranay Please confirm.

Reply:
This is specific to firmware and software services. This is kept as it is because to create all the requied folders whenever csm_agent starts/restarts .
I made the changes at common file save method which will required every module who refers file upload functioanlity.